### DISCUSSION TITLE
Introduce Delay in equations

### DISCUSSION THREAD

**Discussion by Ravua1992** [metrumresearchgroup/mrgsolve#1053] [Q&A] [answered]
Hello, 

I am trying to set up an "interface model for dosage adjustment connects hematotoxicity to pharmacokinetics". In this model there is a feedback loop  and the influx must not only be amplified but also be delayed. Is there any good starting point to implement such models in mrgsolve ?

![image](https://user-images.githubusercontent.com/55991233/217544078-fabd512e-71fa-4d7e-8262-5576807be4be.png)

https://link.springer.com/article/10.1007/s10928-008-9106-4#Sec2

Thank you

**Comment by FelicienLL** [comment]
Hi, yes, have you heard about the Friberg model ? I would say it is the most standard approach for this. https://pubmed.ncbi.nlm.nih.gov/12488418/

**Reply by smouksassi** [reply, marked as answer]
you can recode a delay differential equation model into a series of ODE,s (Method of Steps)
https://math.stackexchange.com/questions/2983414/solving-a-delay-differential-equation-with-the-method-of-steps
but not all systems can be recoded it depends on states, history functions etc.
Monolix, Phonix NLME and more recently NONMEM  added support for DDE and distributed delays but not mrgsolve where as Felicien pointed out you can use a simplified version of the distributed delay model.
https://pubmed.ncbi.nlm.nih.gov/29368268/
